syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "custom_types.proto";
import "google/protobuf/timestamp.proto";

service CardService {
  rpc GetAll (GetAllRequest) returns (GetAllCardsResponse);
  rpc GetById (GetCardByIdRequest) returns (CardModel);
  rpc GetByAccountId (GetCardsByAccountRequest) returns (GetAllCardsResponse);
}

message CardModel {
  int32 card_id = 1;
  optional int32 account_id = 2;
  string card_number = 3;
  string cvv = 4;
  optional google.protobuf.Timestamp expiry_date = 5;
  optional string status = 6;
}

message GetAllCardsResponse {
  repeated CardModel cards = 1;
}

message GetCardByIdRequest {
  int32 card_id = 1;
}

message GetCardsByAccountRequest {
  int32 account_id = 1;
}
